src/app/reporting/Engine.cpp

```diff
@@ -290,21 +291,92 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeReportIBs(ReportDataMessage::Bu
     return err;
 }
 
+CHIP_ERROR Engine::RemoveUnaccessableEventPaths(TLV::TLVWriter & aWriter, bool & aItemRemoved, ReadHandler * apReadHandler)
+{
+    CHIP_ERROR err                                                      = CHIP_NO_ERROR;
+    ObjectList<EventPathParams> * deletedEventPathList                  = nullptr;
+    ObjectList<EventPathParams> ** previousIteratedEventPathListNextPtr = &(apReadHandler->mpEventPathList);
+
+    aItemRemoved = false;
+    for (auto current = apReadHandler->mpEventPathList; current != nullptr;)
+    {
+        if (current->mValue.HasEventWildcard())
+        {
+            current = current->mpNext;
+            continue;
+        }
+
+        Access::RequestPath requestPath{ .cluster = current->mValue.mClusterId, .endpoint = current->mValue.mEndpointId };
+        ConcreteEventPath path;
+        path.mClusterId                    = current->mValue.mClusterId;
+        path.mEndpointId                   = current->mValue.mEndpointId;
+        path.mEventId                      = current->mValue.mEventId;
```

Comment:
Why not pass those to the `ConcreteEventPath` constructor?

Reply:
done, thanks